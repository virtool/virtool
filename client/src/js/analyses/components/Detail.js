import { get } from "lodash-es";
import React, { useEffect } from "react";
import { Panel } from "react-bootstrap";
import { connect } from "react-redux";
import {
    SubviewHeader,
    SubviewHeaderTitle,
    SubviewHeaderAttribution,
    LoadingPlaceholder,
    NotFound,
    RelativeTime
} from "../../base/index";
import { getTaskDisplayName } from "../../utils/utils";
import { clearAnalysis, getAnalysis } from "../actions";
import AnalysisCache from "./CacheLink";
import NuVsViewer from "./NuVs/Viewer";
import PathoscopeViewer from "./Pathoscope/Viewer";

export const AnalysisDetail = props => {
    useEffect(() => {
        props.getAnalysis(props.match.params.analysisId);
        return props.clearAnalysis;
    }, [props.match.params.analysisId]);

    if (props.error) {
        return <NotFound />;
    }

    if (props.detail === null) {
        return <LoadingPlaceholder />;
    }

    const { detail, sampleName } = props;

    if (!detail.ready) {
        return (
            <Panel>
                <Panel.Body>
                    <LoadingPlaceholder message="Analysis in progress" margin="1.2rem" />
                </Panel.Body>
            </Panel>
        );
    }

    let content;

    if (detail.algorithm === "pathoscope_bowtie") {
        content = <PathoscopeViewer />;
    } else if (detail.algorithm === "nuvs") {
        content = <NuVsViewer />;
    } else {
        return <div>"Unusable analysis detail content")</div>;
    }

    return (
        <div>
            <SubviewHeader>
                <SubviewHeaderTitle>
                    {getTaskDisplayName(detail.algorithm)} for {sampleName}
                    <AnalysisCache />
                </SubviewHeaderTitle>
                <SubviewHeaderAttribution>
                    {detail.user.id} started <RelativeTime time={detail.created_at} />
                </SubviewHeaderAttribution>
            </SubviewHeader>

            {content}
        </div>
    );
};

export const mapStateToProps = state => ({
    detail: state.analyses.detail,
    error: get(state, "errors.GET_ANALYSIS_ERROR", null),
    quality: state.samples.detail.quality,
    sampleName: state.samples.detail.name
});

export const mapDispatchToProps = dispatch => ({
    getAnalysis: analysisId => {
        dispatch(getAnalysis(analysisId));
    },

    clearAnalysis: () => {
        dispatch(clearAnalysis());
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AnalysisDetail);
