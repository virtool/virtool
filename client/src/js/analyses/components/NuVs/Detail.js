import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { filter, map, sortBy } from "lodash-es";
import { Badge, Box } from "../../../base";
import { getActiveHit, getMaxSequenceLength } from "../../selectors";
import { useElementSize } from "../../../utils/hooks";
import NuVsBLAST from "./BLAST";
import NuVsORF from "./ORF";
import NuVsSequence from "./Sequence";
import { NuVsValues } from "./Values";

const StyledNuVsFamilies = styled.div`
    border: 1px solid #ddd;
    border-radius: 3px;
    display: flex;
    margin: 3px 0;

    div {
        padding: 3px 8px;
    }

    div:first-child {
        background-color: #ddd;
        border-right: 1px solid #ddd;
    }
`;

const NuVsFamilies = ({ families }) => (
    <StyledNuVsFamilies>
        <div>Families</div>
        <div>{families.length ? families.join(", ") : "None"}</div>
    </StyledNuVsFamilies>
);

const NuVsLayout = styled.div`
    border: 1px solid #ddd;
    margin-bottom: 15px;

    & > div:nth-child(even) {
        background-color: #f5f5f5;
    }
`;

const NuVsDetailTitle = styled.div`
    margin-bottom: 10px;

    h3 {
        align-items: center;
        display: flex;
        font-size: 14px;
        font-weight: bold;
        justify-content: space-between;
        margin: 0;
    }

    span {
        font-size: 12px;
        font-weight: bold;
    }

    ${Badge} {
        font-size: 12px;
        padding: 3px 7px;
    }
`;

const StyledNuVsDetail = styled(Box)`
    align-items: stretch;
    display: flex;
    flex-direction: column;
    min-height: 500px;
    min-width: 0;
    margin-left: 10px;
`;

export const NuVsDetail = ({ filterORFs, hit, maxSequenceLength }) => {
    const [ref, { width }] = useElementSize();

    if (!hit) {
        return <StyledNuVsDetail>No Hits</StyledNuVsDetail>;
    }

    const { blast, e, families, index, orfs, sequence } = hit;

    let filtered;

    if (filterORFs) {
        filtered = filter(orfs, orf => orf.hits.length);
    }

    filtered = sortBy(filtered || orfs, orf => orf.hits.length).reverse();

    const orfComponents = map(filtered, (orf, index) => (
        <NuVsORF key={index} index={index} {...orf} maxSequenceLength={maxSequenceLength} width={width} />
    ));

    return (
        <StyledNuVsDetail ref={ref}>
            <NuVsDetailTitle>
                <h3>
                    Sequence {index}
                    <Badge>{sequence.length} bp</Badge>
                </h3>
                <NuVsValues e={e} orfCount={orfs.length} />
                <NuVsFamilies families={families} />
            </NuVsDetailTitle>
            <NuVsLayout>
                <NuVsSequence key="sequence" sequence={sequence} width={width} />
                {orfComponents}
            </NuVsLayout>
            <NuVsBLAST sequenceIndex={index} blast={blast} sequence={sequence} />
        </StyledNuVsDetail>
    );
};

const mapStateToProps = state => ({
    filterORFs: state.analyses.filterORFs,
    maxSequenceLength: getMaxSequenceLength(state),
    hit: getActiveHit(state)
});

export default connect(mapStateToProps)(NuVsDetail);
