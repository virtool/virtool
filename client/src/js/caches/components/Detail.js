import React, { useEffect } from "react";
import styled from "styled-components";
import { map } from "lodash-es";
import { connect } from "react-redux";
import { Panel, Table } from "react-bootstrap";
import {
    LoadingPlaceholder,
    SubviewHeader,
    SubviewHeaderTitle,
    SubviewHeaderAttribution,
    RelativeTime
} from "../../base";
import { getCache } from "../actions";
import CacheQuality from "./Quality";

const CacheGeneral = ({ hash, program }) => (
    <Table bordered>
        <tbody>
            <tr>
                <th>Hash</th>
                <td>{hash}</td>
            </tr>
            <tr>
                <th>Trimming Program</th>
                <td>{program}</td>
            </tr>
        </tbody>
    </Table>
);

const CacheParameterKey = styled.th`
    font-family: "Roboto Mono", monospace;
    padding-left: 15px !important;
`;

const CacheParameters = ({ parameters }) => {
    const rowComponents = map(parameters, (value, key) => (
        <tr key={key}>
            <CacheParameterKey>{key}</CacheParameterKey>
            <td>{value}</td>
        </tr>
    ));
    return (
        <Panel>
            <Panel.Heading>Trim Parameters</Panel.Heading>
            <Table bordered>
                <tbody>{rowComponents}</tbody>
            </Table>
        </Panel>
    );
};

const StyledCacheDetail = styled.div`
    overflow-x: hidden;
`;

const CacheDetail = ({ detail, match, sampleName, onGet }) => {
    useEffect(() => onGet(match.params.cacheId), [match.params.cacheId]);

    if (detail === null) {
        return <LoadingPlaceholder />;
    }

    return (
        <StyledCacheDetail>
            <SubviewHeader>
                <SubviewHeaderTitle>Cache for {sampleName}</SubviewHeaderTitle>
                <SubviewHeaderAttribution>
                    Created <RelativeTime time={detail.created_at} />
                </SubviewHeaderAttribution>
            </SubviewHeader>
            <CacheGeneral {...detail} />
            <CacheParameters parameters={detail.parameters} />
            <CacheQuality />
        </StyledCacheDetail>
    );
};

const mapStateToProps = state => {
    return {
        detail: state.caches.detail,
        sampleName: state.samples.detail.name
    };
};

const mapDispatchToProps = dispatch => ({
    onGet: cacheId => {
        dispatch(getCache(cacheId));
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CacheDetail);
