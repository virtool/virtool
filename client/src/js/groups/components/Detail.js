import { filter, includes, map, transform } from "lodash-es";
import React, { useCallback } from "react";
import styled from "styled-components";
import {
    Box,
    BoxGroup,
    BoxGroupHeader,
    BoxGroupSection,
    Button,
    Checkbox,
    Icon,
    Loader,
    NoneFoundSection
} from "../../base";

const EmptyGroupDetail = styled(Box)`
    align-items: center;
    display: flex;
    min-height: 220px;
    justify-content: center;

    i {
        margin-right: 3px;
    }
`;

const PermissionsHeader = styled(BoxGroupHeader)`
    flex-direction: row;
    font-weight: bold;
    justify-content: space-between;
`;

const PermissionsItem = styled(BoxGroupSection)`
    display: flex;
    justify-content: space-between;
    align-items: center;
`;

const StyledGroupDetail = styled.div`
    display: flex;
    flex-direction: column;
`;

export const GroupDetail = ({ group, pending, users, onRemove, onSetPermission }) => {
    let members;

    if (!group) {
        return (
            <EmptyGroupDetail>
                <Icon name="exclamation-circle" /> No groups found.
            </EmptyGroupDetail>
        );
    }

    if (group) {
        members = filter(users, user => includes(user.groups, group.id));
    }

    let memberComponents;

    if (members && members.length) {
        memberComponents = map(members, member => (
            <BoxGroupSection key={member.id} spaced>
                {member.id}
            </BoxGroupSection>
        ));
    } else {
        memberComponents = <NoneFoundSection noun="members" />;
    }

    const permissionComponents = transform(
        group.permissions,
        (result, value, key) => {
            result.push(
                <PermissionsItem key={key} onClick={() => onSetPermission(group.id, key, !value)}>
                    <code>{key}</code>
                    <Checkbox checked={value} />
                </PermissionsItem>
            );

            return result;
        },
        []
    );

    const handleRemove = useCallback(() => onRemove(group.id), [group.id]);

    return (
        <StyledGroupDetail>
            <BoxGroup>
                <PermissionsHeader>
                    <span>Permissions</span>
                    {pending ? <Loader size="16px" color="currentColor" /> : null}
                </PermissionsHeader>
                {permissionComponents}
            </BoxGroup>

            <BoxGroup>
                <BoxGroupHeader>Members</BoxGroupHeader>
                {memberComponents}
            </BoxGroup>

            <Button icon="trash" color="red" onClick={handleRemove}>
                Remove Group
            </Button>
        </StyledGroupDetail>
    );
};
