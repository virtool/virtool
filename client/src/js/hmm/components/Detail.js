import { get, map } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";

import {
    device,
    Badge,
    BoxGroup,
    BoxGroupHeader,
    Label,
    LoadingPlaceholder,
    NotFound,
    Table,
    ViewHeader
} from "../../base";
import { getHmm } from "../actions";
import ClusterMembers from "./ClusterMembers";
import { HMMTaxonomy } from "./Taxonomy";

const ClusterTable = styled(Table)`
    border: none;
    display: flex;
    flex-flow: column;
    height: 330px !important;
    margin: 0;
    width: 100%;

    thead {
        flex: 0 0 auto;
        width: calc(100% - 0.9em);

        th {
            border-bottom: none;
        }
    }

    tbody {
        flex: 1 1 auto;
        display: block;
        overflow-y: auto;
        border-top: 1px solid #dddddd;

        tr {
            width: 100%;
        }

        tr:first-child td {
            border-top: none;
        }
    }

    thead,
    tbody tr {
        display: table;
        table-layout: fixed;
    }
`;

const TaxonomyGrid = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    column-gap: 15px;

    @media (max-width: ${device.tablet}) {
        grid-template-columns: 1fr;
    }

    h5 {
        font-weight: bold;
    }
`;

export class HMMDetail extends React.Component {
    componentDidMount() {
        this.props.onGet(this.props.match.params.hmmId);
    }

    render() {
        if (this.props.error) {
            return <NotFound />;
        }

        if (this.props.detail === null) {
            return <LoadingPlaceholder margin="130px" />;
        }

        const names = map(this.props.detail.names, (name, index) => (
            <Label key={index} spaced>
                {name}
            </Label>
        ));
        return (
            <div>
                <ViewHeader title={`${this.props.detail.names[0]} - HMMs`}>
                    <strong>{this.props.detail.names[0]}</strong>
                </ViewHeader>

                <Table>
                    <tbody>
                        <tr>
                            <th>Cluster</th>
                            <td>{this.props.detail.cluster}</td>
                        </tr>

                        <tr>
                            <th>Best Definitions</th>
                            <td>{names}</td>
                        </tr>

                        <tr>
                            <th>Length</th>
                            <td>{this.props.detail.length}</td>
                        </tr>

                        <tr>
                            <th>Mean Entropy</th>
                            <td>{this.props.detail.mean_entropy}</td>
                        </tr>

                        <tr>
                            <th>Total Entropy</th>
                            <td>{this.props.detail.total_entropy}</td>
                        </tr>
                    </tbody>
                </Table>

                <BoxGroup>
                    <BoxGroupHeader>
                        <h2>
                            Cluster Members <Badge>{this.props.detail.entries.length}</Badge>
                        </h2>
                    </BoxGroupHeader>
                    <ClusterTable>
                        <thead>
                            <tr>
                                <th>Accession</th>
                                <th>Name</th>
                                <th>Organism</th>
                            </tr>
                        </thead>
                        <tbody>
                            <ClusterMembers />
                        </tbody>
                    </ClusterTable>
                </BoxGroup>

                <TaxonomyGrid>
                    <HMMTaxonomy title="Families" counts={this.props.detail.families} />
                    <HMMTaxonomy title="Genera" counts={this.props.detail.genera} />
                </TaxonomyGrid>
            </div>
        );
    }
}

export const mapStateToProps = state => ({
    error: get(state, "errors.GET_HMM_ERROR", null),
    detail: state.hmms.detail
});

export const mapDispatchToProps = dispatch => ({
    onGet: hmmId => {
        dispatch(getHmm(hmmId));
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(HMMDetail);
