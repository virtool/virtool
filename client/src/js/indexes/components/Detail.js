import { get } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import {
    LoadingPlaceholder,
    NarrowContainer,
    NotFound,
    RelativeTime,
    SubviewHeader,
    SubviewHeaderAttribution,
    SubviewHeaderTitle
} from "../../base";
import { getIndex, getIndexHistory } from "../actions";
import IndexGeneral from "./General";

export class IndexDetail extends React.Component {
    componentDidMount() {
        this.props.onGetIndex(this.props.match.params.indexId);
        this.props.onGetChanges(this.props.match.params.indexId, 1);
    }

    render() {
        if (this.props.error) {
            return <NotFound />;
        }

        if (this.props.detail === null || this.props.refDetail === null) {
            return <LoadingPlaceholder />;
        }

        const { version, created_at, user } = this.props.detail;

        return (
            <div>
                <SubviewHeader>
                    <SubviewHeaderTitle>Index {version}</SubviewHeaderTitle>
                    <SubviewHeaderAttribution>
                        {user.id} built <RelativeTime time={created_at} />
                    </SubviewHeaderAttribution>
                </SubviewHeader>

                <NarrowContainer>
                    <IndexGeneral />
                </NarrowContainer>
            </div>
        );
    }
}

export const mapStateToProps = state => ({
    error: get(state, "errors.GET_INDEX_ERROR", null),
    detail: state.indexes.detail,
    refDetail: state.references.detail
});

export const mapDispatchToProps = dispatch => ({
    onGetIndex: indexId => {
        dispatch(getIndex(indexId));
    },

    onGetChanges: (indexId, page) => {
        dispatch(getIndexHistory(indexId, page));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(IndexDetail);
