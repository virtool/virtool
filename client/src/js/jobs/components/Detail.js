import { push } from "connected-react-router";
import { get } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import {
    Badge,
    Icon,
    LoadingPlaceholder,
    NotFound,
    Table,
    ViewHeader,
    ViewHeaderAttribution,
    ViewHeaderIcons,
    ViewHeaderTitle
} from "../../base";
import { getTaskDisplayName } from "../../utils/utils";
import { getJob, removeJob } from "../actions";
import JobError from "./Error";
import JobSteps from "./Steps";
import TaskArgs from "./TaskArgs";

const JobDetailBadge = styled(Badge)`
    text-transform: capitalize;
`;

class JobDetail extends React.Component {
    componentDidMount() {
        this.props.getDetail(this.props.match.params.jobId);
    }

    handleClick = () => {
        this.props.onRemove(this.props.detail.id);
    };

    render() {
        if (this.props.error) {
            return <NotFound />;
        }

        if (this.props.detail === null) {
            return <LoadingPlaceholder />;
        }

        const detail = this.props.detail;

        const latest = detail.status[detail.status.length - 1];

        let color = "green";

        if (latest.state === "running") {
            color = "blue";
        }

        if (latest.state === "error" || latest.state === "cancelled") {
            color = "red";
        }

        const taskName = getTaskDisplayName(detail.task);

        return (
            <div>
                <ViewHeader title={taskName}>
                    <ViewHeaderTitle>
                        {taskName} <JobDetailBadge color={color}>{latest.state}</JobDetailBadge>
                        <ViewHeaderIcons>
                            <Icon color="red" name="trash" style={{ fontSize: "18px" }} onClick={this.handleClick} />
                        </ViewHeaderIcons>
                    </ViewHeaderTitle>
                    <ViewHeaderAttribution time={detail.status[0].timestamp} user={detail.user.id} />
                </ViewHeader>

                <Table>
                    <tbody>
                        <tr>
                            <th>Cores / Memory</th>
                            <td>
                                {detail.proc} CPUs / {detail.mem} GB
                            </td>
                        </tr>
                    </tbody>
                </Table>

                <h4>
                    <strong>Task Arguments</strong>
                </h4>

                <TaskArgs taskType={detail.task} taskArgs={detail.args} />

                <JobSteps />

                <JobError error={latest.error} />
            </div>
        );
    }
}

const mapStateToProps = state => ({
    error: get(state, "errors.GET_JOB_ERROR", null),
    detail: state.jobs.detail
});

const mapDispatchToProps = dispatch => ({
    getDetail: jobId => {
        dispatch(getJob(jobId));
    },

    onRemove: jobId => {
        dispatch(removeJob(jobId));
        dispatch(push("/jobs"));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(JobDetail);
