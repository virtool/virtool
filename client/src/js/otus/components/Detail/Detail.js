import { get } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { Link, Redirect, Route, Switch } from "react-router-dom";
import styled from "styled-components";
import { getFontWeight } from "../../../app/theme";
import {
    Icon,
    LoadingPlaceholder,
    NotFound,
    TabLink,
    Tabs,
    ViewHeader,
    ViewHeaderIcons,
    ViewHeaderTitle
} from "../../../base";
import { checkRefRight } from "../../../utils/utils";
import { getOTU, showEditOTU, showRemoveOTU } from "../../actions";
import EditOTU from "./Edit";
import IsolateEditor from "./Editor";
import General from "./General";
import History from "./History/History";
import AddIsolate from "./Isolates/Add";
import RemoveOTU from "./Remove";
import Schema from "./Schema/Schema";

const OTUSection = () => (
    <div>
        <General />
        <IsolateEditor />
        <AddIsolate />
    </div>
);

const OTUDetailTitle = styled(ViewHeaderTitle)`
    align-items: baseline;
    display: flex;

    small {
        color: ${props => props.theme.color.greyDark};
        font-weight: 600;
        margin-left: 7px;

        em {
            font-weight: normal;
        }
    }
`;

const OTUDetailSubtitle = styled.p`
    font-size: ${props => props.theme.fontSize.md};
    margin-top: 5px;

    strong {
        font-weight: ${getFontWeight("thick")};
    }
`;

class OTUDetail extends React.Component {
    componentDidMount() {
        this.props.getOTU(this.props.match.params.otuId);
    }

    render = () => {
        if (this.props.error) {
            return <NotFound />;
        }

        if (this.props.detail === null || this.props.detail.id !== this.props.match.params.otuId) {
            return <LoadingPlaceholder />;
        }

        const refId = this.props.detail.reference.id;
        const { id, name, abbreviation } = this.props.detail;

        let modifyOTUComponents;

        let segmentComponent;
        if (this.props.dataType !== "barcode") {
            segmentComponent = <TabLink to={`/refs/${refId}/otus/${id}/schema`}>Schema</TabLink>;
        }

        let iconButtons;

        if (this.props.canModify) {
            iconButtons = (
                <ViewHeaderIcons>
                    <Icon
                        key="edit-icon"
                        color="orange"
                        name="pencil-alt"
                        tip="Edit OTU"
                        tipPlacement="left"
                        onClick={this.props.showEdit}
                    />
                    <Icon
                        key="remove-icon"
                        color="red"
                        name="trash"
                        tip="Remove OTU"
                        tipPlacement="left"
                        onClick={this.props.showRemove}
                    />
                </ViewHeaderIcons>
            );

            modifyOTUComponents = (
                <div>
                    <EditOTU otuId={id} name={name} abbreviation={abbreviation} />
                    <RemoveOTU id={id} name={name} history={this.props.history} />
                </div>
            );
        }

        return (
            <React.Fragment>
                <ViewHeader title={name}>
                    <OTUDetailTitle>
                        {name} <small>{abbreviation || <em>No Abbreviation</em>}</small>
                        {iconButtons}
                    </OTUDetailTitle>
                    <OTUDetailSubtitle>
                        <strong>From Reference / </strong>
                        <Link to={`/refs/${this.props.detail.reference.id}`}>{this.props.refName}</Link>
                    </OTUDetailSubtitle>
                </ViewHeader>

                <Tabs>
                    <TabLink to={`/refs/${refId}/otus/${id}/otu`}>OTU</TabLink>
                    {segmentComponent}
                    <TabLink to={`/refs/${refId}/otus/${id}/history`}>History</TabLink>
                </Tabs>

                {modifyOTUComponents}

                <Switch>
                    <Redirect from="/refs/:refId/otus/:otuId" to={`/refs/${refId}/otus/${id}/otu`} exact />
                    <Route path="/refs/:refId/otus/:otuId/otu" component={OTUSection} />
                    <Route path="/refs/:refId/otus/:otuId/history" component={History} />
                    <Route path="/refs/:refId/otus/:otuId/schema" component={Schema} />
                </Switch>
            </React.Fragment>
        );
    };
}

const mapStateToProps = state => {
    return {
        error: get(state, "errors.GET_OTU_ERROR", null),
        detail: state.otus.detail,
        refName: state.references.detail.name,
        canModify: !get(state, "references.detail.remotes_from") && checkRefRight(state, "modify_otu"),
        dataType: state.references.detail.data_type
    };
};

const mapDispatchToProps = dispatch => ({
    getOTU: otuId => {
        dispatch(getOTU(otuId));
    },

    showEdit: () => {
        dispatch(showEditOTU());
    },

    showRemove: () => {
        dispatch(showRemoveOTU());
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(OTUDetail);
