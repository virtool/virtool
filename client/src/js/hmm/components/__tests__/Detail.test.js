import { HMMDetail, mapStateToProps, mapDispatchToProps } from "../Detail";

describe("<HMMDetail />", () => {
    let props;
    beforeEach(() => {
        props = {
            onGet: jest.fn(),
            error: false,
            detail: {
                names: ["foo"],
                cluster: "bar",
                mean_entropy: "baz",
                total_entropy: "boo",
                entries: ["Foo"],
                families: "Bar",
                genera: "Baz"
            },
            match: {
                params: {
                    hmmId: "bar"
                }
            }
        };
    });
    it("should render", () => {
        const wrapper = shallow(<HMMDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should return NotFound when [ props.error=true]", () => {
        props.error = true;
        const wrapper = shallow(<HMMDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("should return LoadingPlaceholder when [ props.detail=null]", () => {
        props.detail = null;
        const wrapper = shallow(<HMMDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it("componentDidMount() should call onGet", () => {
        shallow(<HMMDetail {...props} />);
        expect(props.onGet).toHaveBeenCalledWith("bar");
    });
});

describe("mapStateToProps()", () => {
    const state = {
        errors: {
            GET_HMM_ERROR: true
        },
        hmms: {
            detail: "foo"
        }
    };
    it("mapStateToProps() should return props", () => {
        const props = mapStateToProps(state);
        expect(props).toEqual({
            error: true,
            detail: "foo"
        });
    });
});

describe("mapDispatchToProps()", () => {
    const dispatch = jest.fn();

    it("mapDispatchToProps() should return onGet() in props", () => {
        const props = mapDispatchToProps(dispatch);
        props.onGet("foo");
        expect(dispatch).toHaveBeenCalledWith({ hmmId: "foo", type: "GET_HMM_REQUESTED" });
    });
});
