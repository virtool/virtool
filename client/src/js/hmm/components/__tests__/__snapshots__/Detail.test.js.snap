// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HMMDetail /> should render 1`] = `
<div>
  <ViewHeader
    title="foo - HMMs"
  >
    <strong>
      foo
    </strong>
  </ViewHeader>
  <Table>
    <tbody>
      <tr>
        <th>
          Cluster
        </th>
        <td>
          bar
        </td>
      </tr>
      <tr>
        <th>
          Best Definitions
        </th>
        <td>
          <Label
            key="0"
            spaced={true}
          >
            foo
          </Label>
        </td>
      </tr>
      <tr>
        <th>
          Length
        </th>
        <td />
      </tr>
      <tr>
        <th>
          Mean Entropy
        </th>
        <td>
          baz
        </td>
      </tr>
      <tr>
        <th>
          Total Entropy
        </th>
        <td>
          boo
        </td>
      </tr>
    </tbody>
  </Table>
  <Box__BoxGroup>
    <Box__BoxGroupHeader>
      <h2>
        Cluster Members 
        <Badge>
          1
        </Badge>
      </h2>
    </Box__BoxGroupHeader>
    <Detail__ClusterTable>
      <thead>
        <tr>
          <th>
            Accession
          </th>
          <th>
            Name
          </th>
          <th>
            Organism
          </th>
        </tr>
      </thead>
      <tbody>
        <Connect(ClusterMembers) />
      </tbody>
    </Detail__ClusterTable>
  </Box__BoxGroup>
  <Detail__TaxonomyGrid>
    <HMMTaxonomy
      counts="Bar"
      title="Families"
    />
    <HMMTaxonomy
      counts="Baz"
      title="Genera"
    />
  </Detail__TaxonomyGrid>
</div>
`;

exports[`<HMMDetail /> should return LoadingPlaceholder when [ props.detail=null] 1`] = `
<LoadingPlaceholder
  margin="130px"
/>
`;

exports[`<HMMDetail /> should return NotFound when [ props.error=true] 1`] = `<NotFound />`;
